src/appmixer/google/gmail/GetAttachment/GetAttachment.js

```diff
@@ -1,33 +1,19 @@
 'use strict';
-const GoogleApi = require('googleapis');
-const commons = require('../../google-commons');
-const { promisify } = require('util');
-
-// GoogleApi initialization & promisify of some api function for convenience
-const gmail = GoogleApi.gmail('v1');
+const emailCommons = require('../gmail-commons');
 
 module.exports = {
 
     async receive(context) {
-
         const { messageId, attachmentId, fileName } = context.messages.in.content;
-        const getAttachments = promisify(
-            gmail.users.messages.attachments.get.bind(
-                gmail.users.messages.attachments.get
-            )
-        );
-        const attachment = await getAttachments({
-            auth: commons.getOauth2Client(context.auth),
-            userId: 'me',
-            quotaUser: context.auth.userId,
-            id: attachmentId,
-            messageId
+        const endpoint = `/users/me/messages/${messageId}/attachments/${attachmentId}`;
+        const attachment = await emailCommons.callEndpoint(context, endpoint, {
+            method: 'GET',
+            headers: { 'Content-Type': 'application/json' }
         });
 
-        const buffer = Buffer.from(attachment.data, 'base64');
+        const buffer = Buffer.from(attachment.data.data, 'base64');
```

Comment:
it's been here already, however we need to fix this, if possible. We should properly utilize streams working with files. The current versions loads the whole file into memory, but we should stream the file directly from the Google endpoint to our DB. Please investigate this

Reply:
The only way files are imported from google endpoint is in the form of a base64 encoded string, the sum of total files/attachments size on an email can be no more than 25mb.